src/main/java/com/synopsys/integration/detect/tool/detector/inspectors/nuget/NugetInspectorExecutableLocator.java

```diff
@@ -0,0 +1,55 @@
+package com.synopsys.integration.detect.tool.detector.inspectors.nuget;
+
+import java.io.File;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.synopsys.integration.detect.configuration.DetectInfo;
+import com.synopsys.integration.detectable.detectable.exception.DetectableException;
+import com.synopsys.integration.util.OperatingSystemType;
+
+public class NugetInspectorExecutableLocator {
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+    private final DetectInfo detectInfo;
+
+    public NugetInspectorExecutableLocator(DetectInfo detectInfo) {
+        this.detectInfo = detectInfo;
+    }
+
+    @Nullable
+    public File findExecutable(File extractedZip) throws DetectableException {
+        return findExecutable(extractedZip, determineExecutableNameFromOS());
+    }
+
+    @NotNull
+    public String determineExecutableNameFromOS() {
+        if (detectInfo.getCurrentOs() == OperatingSystemType.WINDOWS) {
+            return "detect-nuget-inspector.exe";
+        } else if (detectInfo.getCurrentOs() == OperatingSystemType.MAC) {
+            return "detect-nuget-inspector";
```

Comment:
what's the point of this else-if?

Reply:
Makes me feel warm and fuzzy. I dropped it, I just liked the explicit handling of mac, I'd have done else if == LINUX if it would have let me exhaustively handle an enum.